src/blockdev.rs

```diff
@@ -166,22 +167,32 @@ impl Disk {
     /// Get a handle to the set of device nodes for individual partitions
     /// of the device.
     pub fn get_partition_table(&self) -> Result<Box<dyn PartTable>> {
-        if self.is_dm_device()? {
+        if self.is_dm_device() {
             Ok(Box::new(PartTableKpartx::new(&self.path)?))
         } else {
             Ok(Box::new(PartTableKernel::new(&self.path)?))
         }
     }
 
-    fn is_dm_device(&self) -> Result<bool> {
-        let canon_path = Path::new(&self.path)
-            .canonicalize()
-            .with_context(|| format!("canonicalizing {}", &self.path))?;
-        // convert back to &str because .starts_with on a Path doesn't mean the same thing
-        let canon_path = canon_path
-            .to_str()
-            .with_context(|| format!("path {} is not UTF-8", canon_path.display()))?;
-        Ok(canon_path.starts_with("/dev/dm-"))
```

Comment:
We still need this part since I think the comment there still applies (see https://doc.rust-lang.org/std/path/struct.Path.html#method.starts_with).

Reply:
No, i've made `path` non public, so it's canonicalized in `new()`